### PR TITLE
Refactors Message query to do persistence - prepares renaming to createMessageStore

### PR DESCRIPTION
**Pull request by martin-lysk**
This PR refactors the persistence logic of loadProject.

- lays groundwork for flagged persistence for https://github.com/opral/inlang-message-sdk/issues/45
- fixes https://github.com/opral/inlang-message-sdk/issues/35   

All persistent logic is moved to the queryMessages. 
1. It adds the Message state management directly to the queryMessages function. This removes the need of nested effects on messages
2. It simplifies the dependencies by injecting the signals that effect the query

This PR does **not** change the naming of the query. We can rename the function to something like "messageStore" - but this would increase the changes required so we decided to split the naming.

### CODE REVIEW

**Comment by jldec** [comment]
Ran a before and after load test with 10k messages - results are much better with this PR 
max memory usage down from ~4GB (the node limit) to ~2GB

### before
Screenshot 2024-05-06 at 18 43 32

### after
Screenshot 2024-05-06 at 18 39 02

File: inlang/source-code/sdk/multi-project-test/project4-dir/project.inlang/messages.json
Comment (jldec) [inline inlang/source-code/sdk/multi-project-test/project4-dir/project.inlang/messages.json:1]: I'll push a separate fix to main to get rid of this annoyance -sorry

Comment (jldec) [reply inlang/source-code/sdk/multi-project-test/project4-dir/project.inlang/messages.json:1]: Done - see https://github.com/opral/monorepo/pull/2714
Please merge main to remove this file.

File: inlang/source-code/sdk/src/createMessagesQuery.test.ts [hunk 1]
```diff
@@ -2,15 +2,72 @@
...
+import type { InlangProject } from "./api.js"
 
 const createChangeListener = async (cb: () => void) => createEffect(cb)
 const nextTick = () => new Promise((resolve) => setTimeout(resolve, 0))
 
-describe("create", () => {
-	it("should create a message", () => {
-		const query = createMessagesQuery(() => [])
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const awaitableCreateMessageQuery = async (_messages: () => Message[]) => {
```
Comment (jldec) [inline inlang/source-code/sdk/src/createMessagesQuery.test.ts:16]: not usually a fan of naming with prefix like "awaitable" but in this case it clarifies the test - so 👍

File: inlang/source-code/sdk/src/createMessagesQuery.ts [hunk 2]
```diff
@@ -3,45 +3,193 @@ import { ReactiveMap } from "./reactivity/map.js"
...
+import type { NodeishFilesystem } from "@lix-js/fs"
+import { onCleanup } from "solid-js"
+import { stringifyMessage } from "./storage/helper.js"
+import { acquireFileLock } from "./persistence/filelock/acquireFileLock.js"
+import _debug from "debug"
+import type { ProjectSettings } from "@inlang/project-settings"
+import { releaseLock } from "./persistence/filelock/releaseLock.js"
+import { PluginLoadMessagesError, PluginSaveMessagesError } from "./errors.js"
+import { humanIdHash } from "./storage/human-id/human-readable-id.js"
+const debug = _debug("sdk:loadProject")
```
Comment (jldec) [inline inlang/source-code/sdk/src/createMessagesQuery.ts:18]: let's change this to _debug("sdk:createMessagesQuery")

File: inlang/source-code/sdk/src/createMessagesQuery.ts [hunk 3]
```diff
@@ -87,7 +239,324 @@ export function createMessagesQuery(
 				defaultAliasIndex.delete(message.alias.default)
 			}
 			index.delete(where.id)
+			messageStates.messageDirtyFlags[where.id] = true
+			scheduleSave()
 			return true
 		},
 	}
 }
+
```
Comment (jldec) [inline inlang/source-code/sdk/src/createMessagesQuery.ts:248]: 🙏 thanks for the comments here 👇

File: inlang/source-code/sdk/src/persistence/filelock/releaseLock.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import { type NodeishFilesystem } from "@lix-js/fs"
+import { debugLock } from "../../loadProject.js"
```
Comment (jldec) [inline inlang/source-code/sdk/src/persistence/filelock/releaseLock.ts:2]: please fix as mentioned in https://github.com/opral/monorepo/pull/2709/#pullrequestreview-2042655176

File: inlang/source-code/sdk/src/persistence/filelock/acquireFileLock.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
+import { type NodeishFilesystem } from "@lix-js/fs"
+import type { NodeishStats } from "@lix-js/fs"
+import { debugLock } from "../../loadProject.js"
```
Comment (jldec) [inline inlang/source-code/sdk/src/persistence/filelock/acquireFileLock.ts:3]: Please fix as mentioned in https://github.com/opral/monorepo/pull/2709/#pullrequestreview-2042655176

**Review by jldec — APPROVED** [review]
LGTM (please make small fixes as requested)

**Comment by martin-lysk** [comment]
@NiklasBuchfink can you check this one in fink briefly? Codewise the review from @jldec should be enough

**Pull request merged**